addons/mail/static/src/core/common/thread_service.js

```diff
@@ -633,6 +633,23 @@ export class ThreadService {
         }
     }
 
+    /**
+     * @param {[number]} partnerIds
+     * @param {boolean} inChatWindow
+     */
+    async createChatOrGroupChatWithSelf(partnerIds, inChatWindow = undefined) {
+        const partners_to = [...new Set([this.store.self.id, ...partnerIds])];
+        if (partners_to.length === 1) {
+            await this.joinChat(partners_to[0]).then((chat) => this.open(chat, inChatWindow));
+        } else if (partners_to.length === 2 && partners_to.includes(this.store.self.id)) {
+            await this.joinChat(
+                partners_to.filter((partnerId) => partnerId !== this.store.self.id)[0]
+            ).then((chat) => this.open(chat, inChatWindow));
+        } else {
+            await this.createGroupChat({ partners_to });
+        }
+    }
+
```

Comment:
I would simply call it `makeChat`.

The direction in the future will be to simplify models so that there are less functions. For example, `open`/ `join` functions should all become `openChat`